src/main/java/seedu/address/logic/commands/AddRemoveTagsCommand.java

```diff
@@ -122,11 +124,28 @@ private static Person addTagsToPerson(ReadOnlyPerson personToEdit, Set<Tag> tags
      * Creates and returns a {@code Person} with the details of {@code personToEdit}
      * edited with the {@code tags} to be removed.
      */
-    private static Person removeTagsFromPerson(ReadOnlyPerson personToEdit, Set<Tag> tags)  {
+    private static Person removeTagsFromPerson(ReadOnlyPerson personToEdit, Set<Tag> tags) throws CommandException {
         assert personToEdit != null;
 
         Set<Tag> personTags = personToEdit.getTags();
-        HashSet<Tag> newTags = new HashSet<Tag>(personTags);
+        boolean doesTagsExist = true;
+        ArrayList<Tag> dontExist = new ArrayList<>();
+        for (Tag t: tags) {
+            if (!personTags.contains(t)) {
+                doesTagsExist = false;
+                dontExist.add(t);
+            }
+        }
+
+        if (!doesTagsExist) {
+            String message = MESSAGE_TAG_DONT_EXIST;
+            for (Tag t: dontExist) {
+                message += " " + t.tagName;
```

Comment:
Use a `StringBuilder` instead for better performance

Reply:
Or actually, maybe abstract this to a method that takes a varargs of Strings and returns the full message? I think can use `String#join` as well: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.CharSequence...-